### DISCUSSION TITLE
icat issue

### DISCUSSION THREAD

**Discussion by anurag3301** [kovidgoyal/kitty#3850] [Q&A]
How to fix this problem, 
![ezgif-4-b2f7ad580fc2](https://user-images.githubusercontent.com/52702259/126431001-ae62a438-9e2f-4fe9-b768-eacb03682d30.gif)
I have this alias
```sh
alias icat='kitty +kitten icat '
```
I want only image files to show when I hit the tab.

**Comment by kovidgoyal** [comment]
Turn on completion for kitty in your shell. See #3848

**Reply by anurag3301** [reply]
Sorry but I wasn't able to understand that😅, can you be more specific. Do I have to add something to my `kitty.conf` or `.zshrc`

**Comment by kovidgoyal** [comment]
https://web.archive.org/web/20210526071044/https://sw.kovidgoyal.net/kitty/#completion-for-kitty

**Reply by anurag3301** [reply]
### I added the following as mentioned in the website
![image](https://user-images.githubusercontent.com/52702259/126449721-c61bf61d-f5b5-4d56-ac5e-68d0e75ea47d.png)
### Still, no change, It's still glitchy and shows unwanted options.
![1](https://user-images.githubusercontent.com/52702259/126449965-a6edc79f-2c38-4c83-8ec6-ef4d8c6d2412.gif)
### My [.zshrc](https://pastebin.com/sR58WZqG)

**Reply by kovidgoyal** [reply]
Whats unwanted about the options? those are the options icat supports.

**Reply by kovidgoyal** [reply]
oh and remove the trailing space in your alias definition

**Reply by anurag3301** [reply]
Actually, I don't want to see the options when I hit the tab, I can just check the man page for that.
And removing the trailing space in the alias didn't do anyting. The glitchiness is still there.
My the why I'm using icat is wrong, how do you use icat.

**Reply by Luflosi** [reply]
@kovidgoyal Did you see the GIFs? It looks pretty broken to me.
@anurag3301 execute `echo $TERM` and check that it outputs xterm-kitty.

**Reply by anurag3301** [reply]
@Luflosi Yep, it does
![image](https://user-images.githubusercontent.com/52702259/126473228-2fda2376-ecd4-4020-8451-27068b75626d.png)

**Comment by kovidgoyal** [comment]
On Wed, Jul 21, 2021 at 03:13:16AM -0700, Luflosi wrote:
> @kovidgoyal Did you see the GIFs? It looks pretty broken to me.
> @anurag3301 execute `echo $TERM` and check that it outputs xterm-kitty.

Yes, hence the pointer to the shell branch originally, or even master
where that issue was fixed (zsh cant deal with the coloring of
completion candidates, so it was removed).

**Reply by Luflosi** [reply]
Ok, by bad.
@anurag3301 Try building from source, see https://sw.kovidgoyal.net/kitty/build/ for instructions. If that doesn't work, we can try the shell integration, as @kovidgoyal suggested.

**Reply by kovidgoyal** [reply]
This is the relevant commit cf5ed54
